lib/captain_hook.rb

```diff
@@ -44,17 +44,19 @@ def hook(
     # Hooks logic part
     ####
     def get_hooks(kind)
-      # Only get hooks from the most specific class that defines them
-      return hooks[kind].values if hooks[kind].any?
+      hook_list = Hash.new { |h, k| h[k] = [] }
```

Comment:
what is this? why it needs to be initialized in this way?

Reply:
This is a hash where we’ll store and organize the hooks defined by a class and its ancestors. The key is the hook class name, and the value is an array containing all the occurrences of that hook.